SYCL/SpecConstants/2020/vector-convolution-demo.cpp

```diff
@@ -32,10 +36,21 @@ coeff_struct_t get_coefficient_struct() {
   return {{{{1.0, 2.0, 3.0}, {1.1, 2.1, 3.1}, {1.2, 2.2, 3.2}}}};
 }
 
+// represented in the IR as
+// clang-format off
+// { %struct.coeff_struct_aligned_t { %"class.std::array.0" zeroinitializer, [28 x i8] undef } }
+//                                                                           ~ padding ~
+// clang-format on
+coeff_struct_aligned_t get_coefficient_struct_aligned() {
+  return {{{{1.0, 2.0, 3.0}, {1.1, 2.1, 3.1}, {1.2, 2.2, 3.2}}}};
+}
+
 constexpr specialization_id<coeff_t> coeff_id;
 
 constexpr specialization_id<coeff_struct_t> coeff_struct_id;
 
+constexpr specialization_id<coeff_struct_aligned_t> coeff_struct_aligned_id;
```

Comment:
It may make the test look a little odd, but would it make sense to have two of these? In that case we are sure that one will appear before another spec constant and we can be sure that the calculated offset isn't affected by the padding.

Reply:
@steffenlarsen Thank you for the comment. I'm sorry, maybe I've got your comment wrong. Should I add another spec constant with the same type as `specialization_id<coeff_struct_aligned_t>`, or just remove the `constexpr specialization_id<coeff_struct_t> coeff_struct_id;` one because it has no padding and if the constant with padding has been processed well, the constant without padding should also be processed?

Reply:
My worry is that the changes to sycl-post-link could make it difficult for the runtime to correctly calculate the sizes and offsets of the specialization constants with padding. If the padded specialization constant is the last specialization constant we would likely not see this as there is no following specialization constant and as such the amount of padding does not matter to the runtime. Adding an extra specialization constant with `specialization_id<coeff_struct_aligned_t>` would ensure at least one of them isn't the last specialization constant, right?

Reply:
@steffenlarsen Thank you very much for the explanation, it's crystal clear now. I've added a new specialization constant after `coeff_struct_aligned_id` but I see locally the test failed (in fact, it failed even with one specialization constant containing padding). The test works!